### PR TITLE
add hostname lookup

### PR DESCRIPTION
**Pull request by twouters**
I've hacked in an extra hostname header.
Not sure if it's useful for you and/or done the right way. Comments welcome :-)

I haven't been able to test with punycode and multiple PTR records yet.

I'm also not very happy with the changes in apt_test.go, but I don't know another way to predict the correct hostname.

### CODE REVIEW

**Comment by martinp** [comment]
I can see why this might be useful, but at the current request rate it would cause too many lookups. Thanks though.

**Comment by martinp** [comment]
I suppose I could add it as an optional feature, controlled with a flag. What do you think?

**Comment by twouters** [comment]
That would be great, if you want I can update the pull request

**Comment by martinp** [comment]
Sure. Add a flag for enabling this feature, that defaults to false.

**Comment by twouters** [comment]
I've added a [-r|--reverse] flag - I first opted for [-h|--hostname] but I didn't want to use -h flag.
I also removed the hostname lookup from the tests.

File: api/api.go [hunk 1]
```diff
@@ -30,6 +31,7 @@ var cliUserAgentExp = regexp.MustCompile("^(?i)(curl|wget|fetch\\slibfetch)\\/.*
 type API struct {
 	db       *geoip2.Reader
 	CORS     bool
+	Reverse  bool
```
Comment (martinp) [inline api/api.go]: Can you change this to `ReverseLookup`? Same for the field in `opts`.

**Comment by twouters** [comment]
Changed Reverse to ReverseLookup

**Comment by martinp** [comment]
Thanks. Can you also run gofmt on the changed files?

**Comment by twouters** [comment]
reformatted as requested

**Comment by martinp** [comment]
Thanks!

**Pull request merged**